remote_attestation/java/src/com/google/oak/remote_attestation/Message.java

```diff
@@ -327,6 +329,58 @@ public static ClientIdentity deserialize(byte[] input)
     }
   }
 
+  /** Message containing data encrypted using a session key. */
+  public static class EncryptedData {
+    /** Message header. */
+    private final byte header;
```

Comment:
Isn't the point of this header that it is part of the high-level framing? If so, I would expect to see a class that represents that level of abstraction, basically a header byte, and then payload as bytes, and in this case the payload is the nonce + data.

Reply:
That's not a usable definition; look at your own code, it is basically already parsing the first byte and then based on that deciding how to parse the rest, so there is definitely a higher level framing already implemented, and I think that structure should be reflected more clearly in the code and documentation (e.g. a Frame struct). As another point, it does not make sense if you have a ClientHello frame to get its header; of course its header is the header for the ClientHello frame. (as usual, it's fine to leave a TODO, it does not need to be implemented now)

Reply:
Created an issue: https://github.com/project-oak/oak/issues/2295